### ISSUE TITLE
use "pip install qiling" an error occurred

### ISSUE THREAD

**Issue by stonedreamforest** [qilingframework/qiling#320]
***Describe the bug**
```
C:\Users\test123>pip install qiling
Collecting qiling
  Using cached https://files.pythonhosted.org/packages/99/30/a2d04a3b508e199d60a98f9dbdb3fec3bab514545675a6a259329fab2db6/qiling-1.0-py3-none-any.whl
Collecting pefile>=2019.4.18 (from qiling)
  Using cached https://files.pythonhosted.org/packages/36/58/acf7f35859d541985f0a6ea3c34baaefbfaee23642cf11e85fe36453ae77/pefile-2019.4.18.tar.gz
Collecting keystone-engine>=0.9.1.post3 (from qiling)
  Using cached https://files.pythonhosted.org/packages/9a/fc/ed0d3f46921bfaa612d9e8ce8313f99f4149ecf6635659510220c994cb72/keystone-engine-0.9.1-3.tar.gz
Collecting unicorn>=1.0.2rc3 (from qiling)
  Downloading https://files.pythonhosted.org/packages/b5/62/70df0a455e1f94fc2e1cadd7ca3a9fc7ab159fb77f8fb529ffac94e0c0e9/unicorn-1.0.2rc4-py2.py3-none-win_amd64.whl (1.7MB)
     |████████████████████████████████| 1.7MB 3.3MB/s
Collecting python-registry>=1.3.1 (from qiling)
  Using cached https://files.pythonhosted.org/packages/f1/0f/2605f98bb9c7ef92f926fdf3295583363fe42fc6e8f64cf38b24a029e767/python_registry-1.3.1-py3-none-any.whl
Collecting capstone>=4.0.1 (from qiling)
  Downloading https://files.pythonhosted.org/packages/6a/71/d409f50f7cf7235513f3e636bb1aeb53d729106c1c77495c48fc09c41a54/capstone-4.0.2-py2.py3-none-win_amd64.whl (896kB)
     |████████████████████████████████| 901kB 3.2MB/s
Collecting future (from pefile>=2019.4.18->qiling)
  Using cached https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz
Collecting enum-compat (from python-registry>=1.3.1->qiling)
  Using cached https://files.pythonhosted.org/packages/55/ae/467bc4509246283bb59746e21a1a2f5a8aecbef56b1fa6eaca78cd438c8b/enum_compat-0.0.3-py3-none-any.whl
Collecting unicodecsv (from python-registry>=1.3.1->qiling)
  Using cached https://files.pythonhosted.org/packages/6f/a4/691ab63b17505a26096608cc309960b5a6bdf39e4ba1a793d5f9b1a53270/unicodecsv-0.14.1.tar.gz
Installing collected packages: future, pefile, keystone-engine, unicorn, enum-compat, unicodecsv, python-registry, capstone, qiling
  Running setup.py install for future ... done
  Running setup.py install for pefile ... done
  Running setup.py install for keystone-engine ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\test123\appdata\local\programs\python\python38\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-p0zes1p1\\keystone-engine\\setup.py'"'"'; __file__='"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-p0zes1p1\\keystone-engine\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\test123\AppData\Local\Temp\pip-record-1dul0puy\install-record.txt' --single-version-externally-managed --compile
         cwd: C:\Users\test123\AppData\Local\Temp\pip-install-p0zes1p1\keystone-engine\
    Complete output (50 lines):
    c:\users\test123\appdata\local\programs\python\python38\lib\site-packages\setuptools\dist.py:471: UserWarning: Normalizing '0.9.1-3' to '0.9.1.post3'
      warnings.warn(
    running install
    running build
    running build_py
    creating build
    creating build\lib
    creating build\lib\keystone
    copying keystone\arm64_const.py -> build\lib\keystone
    copying keystone\arm_const.py -> build\lib\keystone
    copying keystone\hexagon_const.py -> build\lib\keystone
    copying keystone\keystone.py -> build\lib\keystone
    copying keystone\keystone_const.py -> build\lib\keystone
    copying keystone\mips_const.py -> build\lib\keystone
    copying keystone\ppc_const.py -> build\lib\keystone
    copying keystone\sparc_const.py -> build\lib\keystone
    copying keystone\systemz_const.py -> build\lib\keystone
    copying keystone\x86_const.py -> build\lib\keystone
    copying keystone\__init__.py -> build\lib\keystone
    running build_clib
    running custom_build_clib
    building 'keystone' library
    '..' 不是内部或外部命令，也不是可运行的程序
    或批处理文件。
    running install_lib
    creating c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\arm64_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\arm_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\hexagon_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\keystone.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\keystone_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\mips_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\ppc_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\sparc_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\systemz_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\x86_const.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    copying build\lib\keystone\__init__.py -> c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\arm64_const.py to arm64_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\arm_const.py to arm_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\hexagon_const.py to hexagon_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\keystone.py to keystone.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\keystone_const.py to keystone_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\mips_const.py to mips_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\ppc_const.py to ppc_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\sparc_const.py to sparc_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\systemz_const.py to systemz_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\x86_const.py to x86_const.cpython-38.pyc
    byte-compiling c:\users\test123\appdata\local\programs\python\python38\Lib\site-packages\keystone\__init__.py to __init__.cpython-38.pyc
    running install_data
    error: can't copy 'src\build\llvm\lib\libkeystone.so': doesn't exist or not a regular file
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\test123\appdata\local\programs\python\python38\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-p0zes1p1\\keystone-engine\\setup.py'"'"'; __file__='"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-p0zes1p1\\keystone-engine\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\test123\AppData\Local\Temp\pip-record-1dul0puy\install-record.txt' --single-version-externally-managed --compile Check the logs for full command output.
WARNING: You are using pip version 19.2.3, however version 20.1.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

```

**Expected behavior**
to fix it

**Comment by aquynh** [comment]
try with 1.1-alpha1 version, with:

`pip3 install qiling --pre`

**Comment by aquynh** [comment]
note that qiling requires Python3, not Python2. so use **pip3**.

**Comment by stonedreamforest** [comment]
> pip3 install qiling --pre

still get an error
```

C:\Users\test123>pip3 install qiling --pre
Collecting qiling
  Downloading https://files.pythonhosted.org/packages/b7/54/235695e7f18e1307161c9fb874ed1ff92bb91fbf7eb39f3196da6b48e834/qiling-1.1a1-py3-none-any.whl (306kB)
     |████████████████████████████████| 307kB 234kB/s
Collecting keystone-engine>=0.9.1.post3 (from qiling)
  Using cached https://files.pythonhosted.org/packages/9a/fc/ed0d3f46921bfaa612d9e8ce8313f99f4149ecf6635659510220c994cb72/keystone-engine-0.9.1-3.tar.gz
Collecting capstone>=4.0.1 (from qiling)
  Using cached https://files.pythonhosted.org/packages/6a/71/d409f50f7cf7235513f3e636bb1aeb53d729106c1c77495c48fc09c41a54/capstone-4.0.2-py2.py3-none-win_amd64.whl
Collecting python-registry>=1.3.1 (from qiling)
  Using cached https://files.pythonhosted.org/packages/f1/0f/2605f98bb9c7ef92f926fdf3295583363fe42fc6e8f64cf38b24a029e767/python_registry-1.3.1-py3-none-any.whl
Collecting unicorn>=1.0.2rc3 (from qiling)
  Using cached https://files.pythonhosted.org/packages/b5/62/70df0a455e1f94fc2e1cadd7ca3a9fc7ab159fb77f8fb529ffac94e0c0e9/unicorn-1.0.2rc4-py2.py3-none-win_amd64.whl
Requirement already satisfied: pefile>=2019.4.18 in c:\users\test123\appdata\local\programs\python\python38\lib\site-packages (from qiling) (2019.4.18)
Collecting enum-compat (from python-registry>=1.3.1->qiling)
  Using cached https://files.pythonhosted.org/packages/55/ae/467bc4509246283bb59746e21a1a2f5a8aecbef56b1fa6eaca78cd438c8b/enum_compat-0.0.3-py3-none-any.whl
Collecting unicodecsv (from python-registry>=1.3.1->qiling)
  Using cached https://files.pythonhosted.org/packages/6f/a4/691ab63b17505a26096608cc309960b5a6bdf39e4ba1a793d5f9b1a53270/unicodecsv-0.14.1.tar.gz
Requirement already satisfied: future in c:\users\test123\appdata\local\programs\python\python38\lib\site-packages (from pefile>=2019.4.18->qiling) (0.18.2)
Installing collected packages: keystone-engine, capstone, enum-compat, unicodecsv, python-registry, unicorn, qiling
  Running setup.py install for keystone-engine ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\test123\appdata\local\programs\python\python38\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-7h3722xj\\keystone-engine\\setup.py'"'"'; __file__='"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-7h3722xj\\keystone-engine\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\test123\AppData\Local\Temp\pip-record-wao2f2uu\install-record.txt' --single-version-externally-managed --compile
         cwd: C:\Users\test123\AppData\Local\Temp\pip-install-7h3722xj\keystone-engine\
    Complete output (27 lines):
    c:\users\test123\appdata\local\programs\python\python38\lib\site-packages\setuptools\dist.py:471: UserWarning: Normalizing '0.9.1-3' to '0.9.1.post3'
      warnings.warn(
    running install
    running build
    running build_py
    creating build
    creating build\lib
    creating build\lib\keystone
    copying keystone\arm64_const.py -> build\lib\keystone
    copying keystone\arm_const.py -> build\lib\keystone
    copying keystone\hexagon_const.py -> build\lib\keystone
    copying keystone\keystone.py -> build\lib\keystone
    copying keystone\keystone_const.py -> build\lib\keystone
    copying keystone\mips_const.py -> build\lib\keystone
    copying keystone\ppc_const.py -> build\lib\keystone
    copying keystone\sparc_const.py -> build\lib\keystone
    copying keystone\systemz_const.py -> build\lib\keystone
    copying keystone\x86_const.py -> build\lib\keystone
    copying keystone\__init__.py -> build\lib\keystone
    running build_clib
    running custom_build_clib
    building 'keystone' library
    '..' 不是内部或外部命令，也不是可运行的程序
    或批处理文件。
    running install_lib
    running install_data
    error: can't copy 'src\build\llvm\lib\libkeystone.so': doesn't exist or not a regular file
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\test123\appdata\local\programs\python\python38\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-7h3722xj\\keystone-engine\\setup.py'"'"'; __file__='"'"'C:\\Users\\test123\\AppData\\Local\\Temp\\pip-install-7h3722xj\\keystone-engine\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\test123\AppData\Local\Temp\pip-record-wao2f2uu\install-record.txt' --single-version-externally-managed --compile Check the logs for full command output.
WARNING: You are using pip version 19.2.3, however version 20.1.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

C:\Users\test123>pip -V
pip 19.2.3 from c:\users\test123\appdata\local\programs\python\python38\lib\site-packages\pip (python 3.8)
```

**Comment by xwings** [comment]
Ah windows pip. I guess we need to fix that in keystone. 

But you are ok to run without keystone.

**Comment by stonedreamforest** [comment]
> Ah windows pip. I guess we need to fix that in keystone.
> 
> But you are ok to run without keystone.

can't import moudle
![image](https://user-images.githubusercontent.com/16742566/83843204-170d2a00-a737-11ea-95c8-9b5abddfd0d4.png)

**Comment by ttimasdf** [comment]
Use WSL or simply Ubuntu VM will save your life. Building python modules is a mess under Windows.

And the linter error is a misconfiguration of VSCode. For example, pylint is unaware of virtualenv or pyenv under default configuration. Editor problem is off topic here.

**Comment by stonedreamforest** [comment]
> Use WSL or simply Ubuntu VM will save your life. Building python modules is a mess under Windows.
> 
> And the linter error is a misconfiguration of VSCode. For example, pylint is unaware of virtualenv or pyenv under default configuration. Editor problem is off topic here.


In fact, I don’t think the installation was successful.
```
PS E:\test\test> pytHon
Python 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> help("modules")

Please wait a moment while I gather a list of all available modules...

Registry            _winapi             heapq               reprlib
__future__          _xxsubinterpreters  hmac                rlcompleter
_abc                abc                 html                runpy      
_ast                aifc                http                sched      
_asyncio            antigravity         idlelib             secrets    
_bisect             appdirs             imaplib             select     
_black_version      argparse            imghdr              selectors  
_blake2             array               imp                 setuptools 
_bootlocale         ast                 importlib           shelve     
_bz2                asynchat            inspect             shlex      
_codecs             asyncio             io                  shutil     
_codecs_cn          asyncore            ipaddress           signal     
_codecs_hk          atexit              itertools           site
_codecs_iso2022     audioop             json                smtpd
_codecs_jp          base64              keystone            smtplib
_codecs_kr          bdb                 keyword             sndhdr
_codecs_tw          binascii            lib2to3             socket
_collections        binhex              libfuturize         socketserver
_collections_abc    bisect              libpasteurize       sqlite3
_compat_pickle      black               linecache           sre_compile
_compression        blackd              locale              sre_constants
_contextvars        blib2to3            logging             sre_parse
_csv                builtins            lzma                ssl
_ctypes             bz2                 mailbox             stat
_ctypes_test        cProfile            mailcap             statistics
_datetime           calendar            marshal             string
_decimal            capstone            math                stringprep
_dummy_thread       cgi                 mccabe              struct
_elementtree        cgitb               mimetypes           subprocess
_functools          chunk               mmap                sunau
_hashlib            cmath               modulefinder        symbol
_heapq              cmd                 msilib              symtable
_imp                code                msvcrt              sys
_io                 codecs              multiprocessing     sysconfig
_json               codeop              mypy                tabnanny
_locale             collections         mypy_extensions     tarfile
_lsprof             colorsys            mypyc               telnetlib
_lzma               compileall          netrc               tempfile
_markupbase         concurrent          nntplib             test
_md5                configparser        nt                  textwrap
_msi                contextlib          ntpath              this
_multibytecodec     contextvars         nturl2path          threading
_multiprocessing    copy                numbers             time
_opcode             copyreg             opcode              timeit
_operator           crypt               operator            tkinter
_osx_support        csv                 optparse            token
_overlapped         ctypes              ordlookup           tokenize
_pickle             curses              os                  toml
_py_abc             dataclasses         parser              trace
_pydecimal          datetime            past                traceback
_pyio               dbm                 pathlib             tracemalloc
_queue              decimal             pathspec            tty
_random             difflib             pdb                 turtle
_sha1               dis                 pefile              turtledemo
_sha256             distutils           peutils             typed_ast
_sha3               doctest             pickle              types
_sha512             dummy_threading     pickletools         typing
_signal             easy_install        pip                 typing_extensions
_sitebuiltins       email               pipes               unicodecsv
_socket             encodings           pkg_resources       unicodedata
_sqlite3            ensurepip           pkgutil             unicorn
_sre                enum                platform            unittest
_ssl                errno               plistlib            urllib
_stat               faulthandler        poplib              uu
_statistics         filecmp             posixpath           uuid
_string             fileinput           pprint              venv
_strptime           flake8              profile             warnings
_struct             fnmatch             pstats              wave
_symtable           formatter           pty                 weakref
_testbuffer         fractions           py_compile          webbrowser
_testcapi           ftplib              pyclbr              winreg
_testconsole        functools           pycodestyle         winsound
_testimportmultiple future              pydoc               wsgiref
_testmultiphase     gc                  pydoc_data          xdrlib
_thread             genericpath         pyexpat             xml
_threading_local    getopt              pyflakes            xmlrpc
_tkinter            getpass             queue               xxsubtype
_tracemalloc        gettext             quopri              zipapp
_warnings           glob                random              zipfile
_weakref            gzip                re                  zipimport
_weakrefset         hashlib             regex               zlib

Enter any module name to get more help.  Or, type "modules spam" to search
for modules whose name or summary contain the string "spam".
```

**Comment by xwings** [comment]
Keystone author is fixing keystone. We need to for for some time.

I tested with WSL2 under Windows and it is awesome. Maybe u can try

**Comment by chfl4gs** [comment]
For Keystone-engine on windows, please use source to compile and installation. Please refer to [COMPILE-WINDOWS.md](https://github.com/keystone-engine/keystone/blob/master/docs/COMPILE-WINDOWS.md)

While fix already submitted [PR#465](https://github.com/keystone-engine/keystone/pull/465), you need to wait for next release to use pip for installation.

**Comment by stonedreamforest** [comment]
the installation is successful now
